### PR TITLE
Data Migration to Support New Discharge Reason Field

### PR DESCRIPTION
**Pull request by Ashesh3**
## Proposed Changes

This PR includes changes for migrating existing data to align with the newly introduced `new_discharge_reason` field in our database schema. The update ensures backward compatibility and seamless integration of the new field across various modules. The modifications are primarily focused on serializers, viewsets, and tests, ensuring that all functionalities related to patient discharge reasons are updated to use the new field. This change is pivotal for enhancing our data handling capabilities and providing more detailed reasons for patient discharge in our system.

### Associated Issue

- https://github.com/coronasafe/care/issues/1739

### Phases for the change

1.  [Schema Changes](https://github.com/coronasafe/care/pull/1749)
2.  **Migrating Data**
3. Moving the temp field back to the original field and deprecate the temp field.

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

**Comment by Ashesh3** [comment]
@rithviknishad @sainak @vigneshhari could you review this PR once?

**Comment by sainak** [comment]
Need a way to notify the call receiver

File: care/facility/api/serializers/patient.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
     BLOOD_GROUP_CHOICES,
     DISEASE_STATUS_CHOICES,
     DiseaseStatusEnum,
+    NewDiseaseReasonEnum,
```
Comment (sainak) [inline care/facility/api/serializers/patient.py:36]: typo?

File: care/facility/api/serializers/patient_consultation.py [hunk 2]
```diff
@@ -111,8 +112,8 @@ class PatientConsultationSerializer(serializers.ModelSerializer):
         queryset=User.objects.all(), required=False, allow_null=True
     )
 
-    discharge_reason = serializers.ChoiceField(
-        choices=DISCHARGE_REASON_CHOICES, read_only=True, required=False
+    new_discharge_reason = serializers.ChoiceField(
+        choices=NEW_DISCHARGE_REASON_CHOICES, read_only=True, required=False
```
Comment (sainak) [inline care/facility/api/serializers/patient_consultation.py:116]:
```suggestion
        choices=NewDischargeReasonEnum.choices, read_only=True, required=False
```

File: care/facility/api/serializers/patient_consultation.py [hunk 3]
```diff
@@ -562,8 +563,8 @@ def validate(self, attrs):
 
 
 class PatientConsultationDischargeSerializer(serializers.ModelSerializer):
-    discharge_reason = serializers.ChoiceField(
-        choices=DISCHARGE_REASON_CHOICES, required=True
+    new_discharge_reason = serializers.ChoiceField(
+        choices=NEW_DISCHARGE_REASON_CHOICES, required=True
```
Comment (sainak) [inline care/facility/api/serializers/patient_consultation.py:567]:
```suggestion
        choices=NewDischargeReasonEnum.choices, required=True
```

File: care/facility/api/viewsets/patient.py [hunk 4]
```diff
@@ -181,6 +184,10 @@ def filter_by_bed_type(self, queryset, name, value):
         field_name="last_consultation__discharge_reason",
         choices=DISCHARGE_REASON_CHOICES,
     )
+    last_consultation_discharge_reason = filters.ChoiceFilter(
+        field_name="last_consultation__discharge_reason",
+        choices=NEW_DISCHARGE_REASON_CHOICES,
```
Comment (sainak) [inline care/facility/api/viewsets/patient.py:189]:
```suggestion
        choices=NewDischargeReasonEnum.choices,
```

**Review by sainak — CHANGES_REQUESTED** [review]

File: care/facility/api/serializers/patient.py (hunk 1 above)
Comment (Ashesh3) [reply care/facility/api/serializers/patient.py]: Yep 😅 Renamed to `NewDischargeReasonEnum`

**Review by sainak — APPROVED** [review]

**Review by rithviknishad — APPROVED** [review]

**Comment by Ashesh3** [comment]
@vigneshhari could you review this?

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**